airflow/jobs/scheduler_job.py

```diff
@@ -1640,11 +1652,13 @@ def _schedule_dag_run(self, dag_run: DagRun, currently_active_runs: int, session
             return 0
 
         if dag.max_active_runs:
-            if currently_active_runs >= dag.max_active_runs:
+            if len(currently_active_runs) >= dag.max_active_runs and \
+               dag_run.execution_date not in currently_active_runs:
                 self.log.info(
```

Comment:
not needed for this PR but should we change this to DEBUG as I remember it was a bit spammy

Reply:
With this change it should happen a lot lot less anyway, but yes, or we fix it properly:

> This approach works around the problem for now, but a better longer term fix for this would be to introduce a "queued" state for DagRuns, and then when manually creating dag runs (or clearing) set it to queued, and only have the scheduler set DagRuns to running, nothing else -- this would mean we wouldn't need to examine active runs in the TI part of the scheduler loop, only in DagRun creation part.